homeassistant/components/notify/__init__.py

```diff
@@ -29,15 +45,43 @@
     check_templates_warn,
 )
 
+# mypy: disallow-any-generics
+
 # Platform specific data
 ATTR_TITLE_DEFAULT = "Home Assistant"
 
+ENTITY_ID_FORMAT = DOMAIN + ".{}"
+
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=10)
+
+_LOGGER = logging.getLogger(__name__)
+
 PLATFORM_SCHEMA = vol.Schema(
     {vol.Required(CONF_PLATFORM): cv.string, vol.Optional(CONF_NAME): cv.string},
     extra=vol.ALLOW_EXTRA,
 )
 
 
+class NotifyDeviceClass(StrEnum):
+    """Device class for notify entities."""
+
+    DIRECT_MESSAGE = "direct_message"
+    DISPLAY = "display"
+    EMAIL = "email"
+
+
+DEVICE_CLASSES_SCHEMA = vol.All(vol.Lower, vol.Coerce(NotifyDeviceClass))
+
+SERVICE_SEND_MESSAGE_SCHEMA = cv.make_entity_service_schema(
+    {
+        vol.Optional(ATTR_MESSAGE): cv.string,
+        vol.Optional(ATTR_TITLE): cv.string,
+        vol.Optional(ATTR_RECIPIENTS): vol.All(cv.ensure_list, [cv.string]),
+        vol.Optional(ATTR_DATA): dict,
```

Comment:
Let's remove data for now.

Reply:
Do we want to keep recipients. It can always become a key later iomo.

Reply:
As discussion on discord we'll remove everything that won't be used in this first MVP PR.

- The message parameter should be required. Integrations that don't have a message to send should not implement this platform.
- Supported features, eg title and recipients, will be removed for now and added later.
- This still leaves ten platforms including the demo integration, that only needs the message parameter, and can potentially be converted to test the new design.

Reply:
Done